### ISSUE TITLE
write with scale="none" not working for floating point data ranging [-1,1]

### ISSUE THREAD

**Issue by tangkk** [WarrenWeckesser/wavio#7]
I found this issue as captioned. Many data to be written are actually floating point data with [-1,1] range. Sometimes when we write them to file we don't want to write them as is and convert them to say 16bit PCM without any normalization.

To solve this we could do something like:
    if scale == "none":
        data = data.clip(outmin, outmax-1).astype(outdtype)
    elif scale == "floatnorm": # data is normalized float number ranging from [-1,1]
        data = _scale_to_sampwidth(data, sampwidth, -1, 1)

See if you could accept this modification. Thanks!

**Comment by WarrenWeckesser** [comment]
@tangkk, thanks for the suggestion.  That looks like the same behavior as setting `scale=(-1, 1)`.  Have you tried that?

**Comment by tangkk** [comment]
> @tangkk, thanks for the suggestion. That looks like the same behavior as setting `scale=(-1, 1)`. Have you tried that?

Yes I believe scale=(-1,1) does the same thing. Thanks for pointing this out since I didn't get it at first: by reading your comments in the "write" function I misunderstood that this is done by setting scale as "none". ("If `scale` is the string "none", then `vmin` and `vmax` are set to
        `outmin` and `outmax`, respectively. This means the data is written
        to the file with no scaling.")
But if I need to convert my float array to 16bit PCM with no scaling, what I should use is actually scale=(-1,1) instead of scale="none", this is a bit strange to me at first...

**Comment by tangkk** [comment]
> @tangkk, thanks for the suggestion. That looks like the same behavior as setting `scale=(-1, 1)`. Have you tried that?

Yes I believe scale=(-1,1) does the same thing. Thanks for pointing this out since I didn't get it at first: by reading your comments in the "write" function I misunderstood that this is done by setting scale as "none". ("If `scale` is the string "none", then `vmin` and `vmax` are set to
        `outmin` and `outmax`, respectively. This means the data is written
        to the file with no scaling.")
But if I need to convert my float array to 16bit PCM with no scaling, what I should use is actually scale=(-1,1) instead of scale="none", this is a bit strange to me at first...

**Comment by WarrenWeckesser** [comment]
The various `scale` options can be confusing.  The option `scale='none'` means that the floating point input values are converted directly to integers, without any shifting or scaling of the input values.  So if the floating point input array is in the range [-1, 1], the converted integer values will be just -1, 0, and 1.  For example, the sequence [0.5, 0.75, 1, 0.6, -0.2, -1.0, -0.9, 0.1, 0.3] would be converted to [0, 0, 1, 0, 0, -1, 0, 0, 0].

> But if I need to convert my float array to 16bit PCM with no scaling

But it sound like you *do* want scaling (at least with my interpretation of what "scaling" means).  You want to scale up the values from [-1, 1] to the 16 bit integer range `[-2**15, 2**15-1]`.  To me, "no scaling" means to convert the floating point values directly to integers, so a value such as 14.35 becomes 14.

I'll work on adding another example to the docstring that shows the use of `scale=(-1, 1)` (or something similar).

**Comment by tangkk** [comment]
Thanks for your reply. I think the solution you provide is a good one. The issue could be closed now...

But I should pointed out that the "no scaling" may have little usage scenario given that the sampwidth is changed, which means if you change "sampwidth" to something other than the original sampwidth, you will almost always need to scale. And if you'd just like to keep the "volume" of the output unchanged, you'll have to know exactly the range of the original sampwidth (e.g. [-1,1] in my case).

**Comment by WarrenWeckesser** [comment]
> But I should pointed out that the "no scaling" may have little usage scenario 

I agree, that option is probably not one that will be used often.

I think what you want is the default, `scale=None`, which, as explained in the docstring, is not the same as `scale="none"`.  The docstring also explains that "By default, the data written to the file is scaled up or down to occupy the full range of the output data type."  So I think not setting the `scale` argument at all should work for you.  Just use the default, and your values in the interval [-1, 1] are scaled up to the limits of the output data type.

For example,  in the following code, `x` is a floating point array with values in [-1, 1].  It is written to a file using `sampwidth=2`.  When the values are read back, we see that they have been expanded to fill the range of signed 16 integers.

```
In [30]: x = np.array([0, 0.5, 1, 0.25, -0.5, -1.0, -0.25])

In [31]: wavio.write("x.wav", x, 44100, sampwidth=2)

In [32]: wavio.read("x.wav").data
Out[32]: 
array([[     0],
       [ 16384],
       [ 32767],
       [  8192],
       [-16384],
       [-32768],
       [ -8192]], dtype=int16)
```
